discretize/CylMesh.py

```diff
@@ -200,48 +763,96 @@ def faceDiv(self):
-            D1 = utils.kron3(utils.speye(self.nCz), utils.speye(self.nCy),
-                             utils.ddx(self.nCx)[:,1:])
-            S = self.r(self.area, 'F', 'Fx', 'V')
+            if self.isSymmetric:
+                D1 = utils.kron3(
+                    utils.speye(self.nCz), utils.speye(self.nCy),
+                    utils.ddx(self.nCx)[:, 1:]
+                )
+            else:
+                D1 = super(CylMesh, self)._faceDivStencilx
+
+            S = self._areaFxFull
             V = self.vol
             self._faceDivx = utils.sdiag(1/V)*D1*utils.sdiag(S)
+
+            if not self.isSymmetric:
+                self._faceDivx = (
+                    self._faceDivx *
+                    self._deflationMatrix('Fx', withHanging=True, asOnes=True).T
+                )
+
         return self._faceDivx
 
     @property
     def faceDivy(self):
         """
         Construct divergence operator in the y component
-        (face-stg to cell-centres).
+        (faces to cell-centres).
         """
-        raise NotImplementedError('Wrapping the utils.ddx is not yet '
-                                  'implemented.')
         if getattr(self, '_faceDivy', None) is None:
-            # TODO: this needs to wrap to join up faces which are
-            # connected in the cylinder
-            D2 = utils.kron3(utils.speye(self.nCz), utils.ddx(self.nCy),
-                            utils.speye(self.nCx))
-            S = self.r(self.area, 'F', 'Fy', 'V')
+            D2 = super(CylMesh, self)._faceDivStencily
+            S = self._areaFyFull #self.r(self.area, 'F', 'Fy', 'V')
             V = self.vol
-            self._faceDivy = utils.sdiag(1/V)*D2*utils.sdiag(S)
+            self._faceDivy = (
+                utils.sdiag(1/V)*D2*utils.sdiag(S) *
+                self._deflationMatrix('Fy', withHanging=True, asOnes=True).T
+            )
         return self._faceDivy
 
     @property
     def faceDivz(self):
         """
         Construct divergence operator in the z component
-        (face-stg to cell-centres).
+        (faces to cell-centres).
         """
         if getattr(self, '_faceDivz', None) is None:
-            D3 = utils.kron3(utils.ddx(self.nCz), utils.speye(self.nCy),
-                             utils.speye(self.nCx))
-            S = self.r(self.area, 'F', 'Fz', 'V')
+            D3 = super(CylMesh, self)._faceDivStencilz
+            S = self._areaFzFull
             V = self.vol
             self._faceDivz = utils.sdiag(1/V)*D3*utils.sdiag(S)
         return self._faceDivz
 
+    # @property
+    # def _cellGradxStencil(self):
+    #     n = self.vnC
+
+    #     if self.isSymmetric:
+    #         G1 = sp.kron(utils.speye(n[2]), ddxCellGrad(n[0], BC))
+    #     else:
+    #         G1 = self._deflationMatrix('Fx').T * utils.kron3(
+    #             utils.speye(n[2]), utils.speye(n[1]), ddxCellGrad(n[0], BC)
+    #         )
+    #     return G1
+
+    @property
```

Comment:
- [ ] TODO: this should raise a `NotImplementedError` right now - it is not tested